src/App/GeoFeatureGroupExtension.cpp

```diff
@@ -421,10 +421,8 @@ bool GeoFeatureGroupExtension::extensionGetSubObject(DocumentObject*& ret,
             }
         }
         if (ret) {
-            if (dot) {
-                ++dot;
-            }
-            if (dot && *dot && !ret->hasExtension(App::LinkBaseExtension::getExtensionClassTypeId())
+            ++dot;
```

Comment:
We reuse `dot` here to mean after-dot-position.
Would it be an option to do:

```suggestion
            const char* afterDot = dot + 1;
```

And then use that variable going forward? It would make the algorithm a bit more clear.

Totally optional though as it was there before your change.

Reply:
I made the change, reread the method, and changed my mind. This method is pretty opaque, and that change doesn't really move the needle. I don't think it's worth making any non-automatic refactorings here until the method is under test, at which point it probably needs a rewrite.